File: front/api/DTO/modelTraining.dto.ts
import RoomLightDTO from "./roomLight.dto";
import { ScratchLayersDTO } from "./scratchLayers.dto";
import { CustomLayersDTO } from "./customLayers.dto";



export default interface ModelTrainingDTO {
  type: "base" | "scratch";
  epochs: number;
  batchSize: number;
  learningRate: number;
  roomList: RoomLightDTO[];
  scratchLayers?: ScratchLayersDTO;
  customLayers?: CustomLayersDTO;
}

File: back/app/model/domain/DTO/modelTrainingDTO.py
from pydantic import BaseModel
from typing import Literal, Optional
from app.room.domain.DTO.roomLightDTO import RoomLightDTO
from app.model.domain.DTO.scratchLayersDTO import ScratchLayersDTO


class ModelTrainingDTO(BaseModel):
    type: Literal["base", "scratch"]
    epochs: int
    batchSize: int
    learningRate: float
    roomList: list[RoomLightDTO]
    scratchLayers: Optional[ScratchLayersDTO] = None
